Fix the Istio CNI doc per the rel-1.1 helm chart decoupling changes (#3537)
istio/istio.io: content/docs/setup/kubernetes/install/cni/index.md

    1.  Refer to the [Hosted Kubernetes Usage](#hosted-kubernetes-usage) section for any non-default
        settings required.

> 1.  Update Helm's local package cache per the instructions in the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/).

Reviewer: ```suggestion
1.  Update Helm's local package cache with the instructions found in the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/). 
    To install using the `helm template`, fetch the `istio-cni` template in addition
    to the other Istio templates you fetched when following [option 1 of the Istio Helm installation procedure](/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template).

```

>     1.  If installing via `helm template` fetch the `istio-cni` template in addition to the other Istio

Reviewer: To help simplify things for readers, I suggest you add this content to the main step. I've left a suggestion that does this for you to be able to quickly implement it. I'm also removing the lines that then become redundant.
```suggestion
```

>         templates fetched per the [Istio Helm installation procedure](/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template)

Reviewer: ```suggestion
```

>         {{< text bash >}}

Reviewer 2: I think you need some kind of note here to indicate the CNI charts are pushed to the same locations that the istio charts are pushed to.   As we discussed directly I don't think you want to get in the version quagmire of which specific location to grab from.   For most people it may not be obvious that the CNI charts are pushed to the same locations.

Author: IMHO, since this is user facing doc they shouldn't care about those release & src repo details.  Doing the `helm repo add istio.io ...` instructions always makes these CNI instructions work.  So, I'm wary to add more complicated info here.

Reviewer: @john-a-joyce We can add the details of the implementation in a separate document at a later date. Adding the details here could confuse and over-complicate the procedure.

        $ helm fetch istio.io/istio-cni --untar --untardir $HOME/istio-fetch
        {{< /text >}}

1.  Create a namespace for the Istio control-plane components:

    {{< text bash >}}
    $ kubectl create namespace istio-system
    {{< /text >}}

1.  Install Istio CNI.

>     1.  Option 1.  Render and apply the Istio CNI Helm template.

Reviewer: Do not use numbered lists for alternative options. The numbers tell readers that they should run one command after the other and that is not the case here. 

```suggestion
    - Option 1.  Render and apply the Istio CNI Helm template.
```

        {{< text bash >}}
        $ helm template $HOME/istio-fetch/istio-cni --name=istio-cni --namespace=istio-system | kubectl apply -f -
        {{< /text >}}

>     1.  Option 2.  Install via Helm and Tiller via `helm install`.

Reviewer: ```suggestion
    - Option 2.  Install via Helm and Tiller via `helm install`.
```

    {{< /tip >}}

1.  Add the `--set istio_cni.enabled=true` setting to the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/) to include the installation of the Istio CNI plugin. For example:

    {{< text bash >}}
    $ helm template install/kubernetes/helm/istio --name istio --namespace istio-system \
      --set istio_cni.enabled=true > $HOME/istio.yaml
    {{< /text >}}

### Example: excluding specific Kubernetes namespaces

The following example configures the Istio CNI plugin to ignore pods in the namespaces `istio-system`,
`foo_ns`, and `bar_ns`.

1.  Create an `istio-system` namespace for the Istio control-plane components.

    {{< text bash >}}
    $ kubectl create namespace istio-system
    {{< /text >}}

1.  Create and apply Istio CNI manifest.  Override the default configuration of the `istio-cni` Helm
    chart's `logLevel` and `excludeNamespaces` parameters:

    {{< text bash >}}
    $ helm template $HOME/istio-fetch/istio-cni --name=istio-cni --namespace=istio-system \
    --set istio-cni.logLevel=info \

>     --set istio-cni.excludeNamespaces={"istio-system,foo_ns,bar_ns"} > $HOME/istio-cni.yaml

Reviewer 2: Do you want to keep the "istio-cni" part of "istio-cni.excludeNamespaces".  I am not sure if it breaks including it,  but it shouldn't be required.

Author: you're right ... will fix

    enable [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) in your cluster.
    *  Note for existing clusters this redeploys the nodes.

1.  Install Istio CNI via Helm including this option

>     `--set cniBinDir=/home/kubernetes/bin`

Reviewer: ```suggestion
    `--set cniBinDir=/home/kubernetes/bin`. For example, the following `helm install`
    command sets the `cniBinDir` value for GKE setups:
```

>     1.  For example, the following `helm install` command sets the `cniBinDir` value for GKE setups:

Reviewer: ```suggestion
```

    enable [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) in your cluster.
    *  Note for existing clusters this redeploys the nodes.

> 1.  Install Istio CNI via Helm including these options

Reviewer 2: You lost the "istio_cni.enabled=true" part of the instructions.  Maybe reword to say "Install Istio CNI via Helm as described above, but also with this option."  Note also use of "these options" in original text is not proper.

Author: I removed the `istio_cni.enabled` since the below refers to the istio-cni param.  Maybe I will add an example it like this:

```
$ helm install istio.io/istio-cni --name=istio-cni --namespace=istio-system --set cniBinDir=/home/kubernetes/bin
```

    enable [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) in your cluster.
    *  Note for existing clusters this redeploys the nodes.

1.  Install Istio CNI via Helm including this option
    `--set cniBinDir=/home/kubernetes/bin`. For example, the following `helm install`
    command sets the `cniBinDir` value for GKE setups:

> 

Author: need to fix up the spacing and newlines here... it rendered funny
